resolve/index.go

```diff
@@ -168,12 +178,7 @@ func (ix *RuleIndex) AddRule(c *config.Config, r *rule.Rule, f *rule.File) {
 		embeds:     embeds,
 		lang:       lang,
 	}
-	if _, ok := ix.labelMap[record.label]; ok {
```

Comment:
This is the one change in this PR... now duplicates will be persisted in the `ix.rules` and only resolved when indexing. I think this is rare enough the extra data being persisted is worth cleanly separating the "source of truth" and indexed data 🤷